include/swift/AST/ExistentialLayout.h

```diff
@@ -70,9 +90,12 @@ struct ExistentialLayout {
   }
   typedef ArrayRefView<Type,ProtocolType*,getProtocolType> ProtocolTypeArrayRef;
 
-  ProtocolTypeArrayRef getProtocols() const {
+  ProtocolTypeArrayRef getProtocols() const & {
     return protocols;
```

Comment:
Can we put the logic in here instead, and avoid having a copy ctor?

Reply:
What logic?